Sources/Remote Logging/Crash Logging/CrashLogging.swift

```diff
@@ -231,6 +235,59 @@ public extension CrashLogging {
         }
     }
 
+    /// Sends an `Event` immediately and triggers a callback on completion
+    func logMessageImmediately(_ message: String, properties: [String: Any]? = nil, level: SentryLevel = .info, callback: @escaping (Result<Bool, Error>) -> Void) throws {
+        var serializer = SentryEventSerializer(dsn: dataProvider.sentryDSN)
+
+        let event = Event(level: level)
+        event.message = SentryMessage(formatted: message)
+        event.extra = properties
+        event.timestamp = Date()
+        event.user = dataProvider.currentUser?.sentryUser
+
+        serializer.add(event: addStackTrace(to: event))
+
+        guard let requestBody = try? serializer.serialize() else {
+            TracksLogError("⛔️ Unable to send errors to Sentry – error could not be serialized. Attempting to schedule delivery for another time.")
+            SentrySDK.capture(event: event)
+            return
+        }
+
+        try submitSerializedEvents(payload: requestBody, callback: callback)
+    }
+
+    /**
+     Writes the error to the Crash Logging system, and includes a stack trace. This method will block the thread until the event is fired.
+
+     - Parameters:
+     - error: The error object
+     - userInfo: A dictionary containing additional data about this error.
+     - level: The level of severity to report in Sentry (`.error` by default)
+    */
+    func logMessageAndWait(_ message: String, properties: [String: Any]? = nil, level: SentryLevel = .info) throws {
+        let semaphore = DispatchSemaphore(value: 0)
+
+        var networkError: Error?
```

Comment:
What's the reasoning for this blocking the thread?

Reply:
Extracted from [logErrorAndWait](https://github.com/Automattic/Automattic-Tracks-iOS/blob/develop/Sources/Remote%20Logging/Crash%20Logging/CrashLogging.swift#L210).

In the `Panic Log` scenario, we want errors to be reported synchronously, in order to:

- Avoid errors storing the crashlog in the filesystem (and thus, loosing the metadata)
- Immediately hit Sentry's endpoint